inc/saiswitch.h

```diff
@@ -3004,6 +3083,26 @@ typedef enum _sai_switch_stat_t
  */
 #define SAI_KEY_HW_PORT_PROFILE_ID_CONFIG_FILE    "SAI_HW_PORT_PROFILE_ID_CONFIG_FILE"
 
+/**
+ * @brief Switch health event callback
+ *
+ * @objects switch_id SAI_OBJECT_TYPE_SWITCH
+ *
+ * @param[in] switch_id Switch Id
+ * @param[in] severity Health event severity
+ * @param[in] timestamp Time and date of receiving the SDK Health event
+ * @param[in] category Category of cause
+ * @param[in] data Data of switch health
+ * @param[in] description Free text with relevant information delivered from SDK event/trap
+ */
+typedef void (*sai_switch_asic_sdk_health_event_notification_fn)(
+        _In_ sai_object_id_t switch_id,
+        _In_ sai_switch_asic_sdk_health_severity_t severity,
+        _In_ sai_timespec_t timestamp,
+        _In_ sai_switch_asic_sdk_health_category_t category,
+        _In_ sai_switch_health_data_t data,
+        _In_ sai_u8_list_t description);
```

Comment:
this probably should be const since contains pointer to data

Reply:
Right now data is passed as value not as pointer
_In_ sai_switch_health_data_t data,
Do you suggest changing to
_In_ const sai_switch_health_data_t *data
?

Reply:
i was talking about sai_u8_list_t, since u8 list has pointer inside, which without const can be modified